src/com/owncloud/android/lib/resources/files/DownloadRemoteFileOperation.java

```diff
@@ -142,18 +142,19 @@ private RemoteOperationResult downloadFile(OwnCloudClient client, File targetFil
                 }
                 if (transferred == totalToTransfer) {  // Check if the file is completed
                     savedFile = true;
-                    Header modificationTime = mGet.getResponseHeader("Last-Modified");
-                    if (modificationTime == null) {
-                        modificationTime = mGet.getResponseHeader("last-modified");
-                    }
+                    final String modificationTime =
+                            mGet.getResponseHeaders().get("Last-Modified") != null
+                            ? mGet.getResponseHeaders().get("Last-Modified")
+                            :  mGet.getResponseHeader("last-modified");
+
                     if (modificationTime != null) {
-                        Date d = WebdavUtils.parseResponseDate(modificationTime.getValue());
+                        final Date d = WebdavUtils.parseResponseDate(modificationTime);
                         mModificationTimestamp = (d != null) ? d.getTime() : 0;
                     } else {
                         Log_OC.e(TAG, "Could not read modification time from response downloading " + mRemotePath);
                     }
 
-                    mEtag = WebdavUtils.getEtagFromResponse(mGet);
+                    // TODO mEtag = WebdavUtils.getEtagFromResponse(mGet);
                     if (mEtag.length() == 0) {
```

Comment:
It always will be zero. Don't you nedd implement TODO?

Reply:
Let me check...

Reply:
Yes, I will implement the TODO